### ISSUE TITLE
Development help wanted!

### ISSUE THREAD

**Issue by dermotduffy** [dermotduffy/frigate-hass-card#1248]
If you're interested in contributing to this project, I'd more than welcome it! The pace of feature requests + bug fixes exceed my ability to keep up, I'd really appreciate some help if someone(s) had the inclination and I'd be happy to code-review and assist to help you get up to speed.

Skills that are likely required:

* A passion for Home Assistant and optionally Frigate :-)
* At least some Javascript and/or CSS experience.

Skills that would really help (but could be picked up if you have the above):

* **Typescript**: This entire project is written in Typescript.
* **SCSS**: All styles are written in SCSS.
* **[vitest](https://vitest.dev/)**: All testing uses vitest.

If you'd like to contribute, feel free to chime in on this issue or contact me on discord (`dermotduffy`) and I could recommend a good starter issue from the backlog.

**Comment by kirsch33** [comment]
@dermotduffy  is it appropriate to create issues relating to dev builds? i dont want to create issues relating to things that are obviously a work in progress which would be annoying, but you've previously mentioned not having an iPhone and relying on others to give good feedback for that platform. Maybe an alternative would be a discussion thread for upcoming releases? Kinda like how Frigate does during their release cycles.

**Comment by dermotduffy** [comment]
> @dermotduffy is it appropriate to create issues relating to dev builds? i dont want to create issues relating to things that are obviously a work in progress which would be annoying, but you've previously mentioned not having an iPhone and relying on others to give good feedback for that platform. Maybe an alternative would be a discussion thread for upcoming releases? Kinda like how Frigate does during their release cycles.

Once a PR is merged into the `dev` branch it is expected to be internally consistent and function reasonably. At that point, issues are just fine and most welcome! Continuous build runs of a PR that is not yet merged are a different story, as there may be more commits to come for the PR.

In general though, the problem is not too few issues, it's too few people fixing them :-)

(Also fine with the discussion thread, but wasn't imagining that card usage was high enough to have a lively discussion forum).

**Comment by duro** [comment]
I also tried to fire up the DevContainer based dev environment, but it appears to have some incompatibilities with a Mac dev environment.

For example:

- It uses some `sed` flags which are not available on the Homebrew version of `sed`
- The Volta dependency apparently is only available for amd64 architectures, and without some re-jiggering, it defaults to arm64 on M1/2 macs.

**Comment by dermotduffy** [comment]
@duro Thanks for trying! Not surprised, all development on this card that I am aware of (and certainly my own) are all on Linux. If you're interested in developing on Mac natively, I'd welcome changes to the setup to make it more platform agnostic -- it's just that no-one has needed / used that.

PS: `volta` is used just to keep the installation of node/yarn consistent. I'm not attached to it if there's a good reason to change from it, but it has been useful at eliminating weirdnesses from different node/yarn versions.

**Comment by groyee** [comment]
Any plans to add support for devcontainer for M1?
